substrate/client/executor/common/src/runtime_blob/runtime_blob.rs

```diff
@@ -192,7 +190,7 @@ impl RuntimeBlob {
 		match self.0 {
 			BlobKind::WebAssembly(raw_module) =>
 				serialize(raw_module).expect("serializing into a vec should succeed; qed"),
-			BlobKind::PolkaVM(ref blob) => blob.as_bytes().to_vec(),
+			BlobKind::PolkaVM(ref blob) => blob.code().to_vec(),
```

Comment:
This is incorrect; `code` doesn't return the original blob, it just returns the code subsection.

What you need to do is in `new` convert the `raw_blob` into an `ArcBytes`, and store this `ArcBytes` along with the `ProgramBlob` in the `PolkaVM` enum variant. Then you can return those `ArcBytes` here.

(NB: This method might actually not be strictly necessary when running with a PolkaVM-based executor, but let's keep it implemented for now, as otherwise we'd have to change the return type to be fallible, and that'd be a bigger change.)

Reply:
OK, will revisit.

Reply:
https://github.com/paritytech/polkadot-sdk/pull/6533/commits/1f888ede52b1280525c8db6e6634532d375f63f9